apps/src/pythonlab/PythonlabView.tsx

```diff
@@ -204,6 +235,10 @@ const PythonlabView: React.FunctionComponent<
       );
     }
     dispatch(submitPredictResponse({appType: 'pythonlab'}));
+
+    // Set a question for the hint AITutor2 in ValidatedInstructions to ask.
+    // It will be passed down via the CodebridgeContext.
+    setAiTutor2HintQuestion("What's wrong with my code, if anything?");
```

Comment:
To clarify, this doesn't actually ask the question right? The user still needs to manually kick off the request? Edit: oh I see there's a `useEffect` in AITutor2UI that automatically kicks off the request

Reply:
Yes.  I'm effectively using this to trigger the ask that's done in `AITutor2UI`.  Curious if there's a better mechanism...

Reply:
Turns out an answer lay [here](https://github.com/code-dot-org/code-dot-org/pull/65737#issuecomment-2896744361), with a custom hook (a function that can use other hooks) that both renders the component and provides a function that can be explicitly called to ask the question, keeping all the state inside the hook.